crates/cast/src/lib.rs

```diff
@@ -163,40 +163,50 @@ pub fn get_network(name: &str) -> Result<Network> {
 pub async fn wait_for_tx(
     provider: &JsonRpcClient<HttpTransport>,
     tx_hash: FieldElement,
+    retries: u8,
 ) -> Result<&str> {
-    'a: while {
-        let receipt = provider
-            .get_transaction_receipt(tx_hash)
-            .await
-            .unwrap_or_else(|_| panic!("Could not get transaction with hash: {tx_hash:#x}"));
-
-        let status = if let Receipt(receipt) = receipt {
-            match receipt {
-                Invoke(receipt) => receipt.status,
-                Declare(receipt) => receipt.status,
-                Deploy(receipt) => receipt.status,
-                DeployAccount(receipt) => receipt.status,
-                L1Handler(receipt) => receipt.status,
-            }
-        } else {
-            continue 'a;
-        };
-
-        match status {
-            TransactionStatus::Pending => {
-                sleep(Duration::from_secs(5));
-                true
-            }
-            TransactionStatus::AcceptedOnL2 | TransactionStatus::AcceptedOnL1 => {
-                return Ok("Transaction accepted")
+    println!("Transaction hash: {tx_hash:#x}");
+
+    let mut retries = retries;
+    'a: loop {
+        sleep(Duration::from_secs(5));
```

Comment:
It could perhaps make sense to try and poll for tx status at least once before sleep

Reply:
I think that would make sense in the case of devnet (and maybe Starknet) in some time but for now all the transactions I've tried are available after (at least) 10 seconds.
However, I think I can change that to be future-proof